sources/packages/api/src/services/student/student.service.ts

```diff
@@ -329,22 +328,22 @@ export class StudentService extends RecordDataModelService<Student> {
       ])
       .innerJoin("student.user", "user")
       .where("user.isActive = true");
-    if (firstName) {
+    if (searchCriteria.firstName) {
       searchQuery.andWhere("user.firstName Ilike :firstName", {
-        firstName: `%${firstName}%`,
+        firstName: `%${searchCriteria.firstName}%`,
       });
     }
-    if (lastName) {
+    if (searchCriteria.lastName) {
       searchQuery.andWhere("user.lastName Ilike :lastName", {
-        lastName: `%${lastName}%`,
+        lastName: `%${searchCriteria.lastName}%`,
       });
     }
-    if (appNumber) {
+    if (searchCriteria.appNumber) {
       searchQuery
         .andWhere("application.applicationNumber Ilike :appNumber")
         .andWhere("application.applicationStatus != :overwrittenStatus")
```

Comment:
I know, its not part of the PR. shouldn't `.andWhere("application.applicationStatus != :overwrittenStatus")` be outside the `if`

Reply:
It seems that if the application will be searched then look only into the not overwrittenStatus. Either way, it seems weird t me too.
@guru-aot can you refresh our memory here 😉

Reply:
This is a student search and, if the user searches with just application number, then we search for the application which are all not in the overwritten status. If we have it outside the if, then the search happens with respect to user table , so the application status is irrelevant.